content/shmem_barrier.tex

```diff
@@ -54,17 +55,17 @@
     \acp{PE} in the active set on the default context are complete before returning.
     
     The  same  \VAR{pSync} array may be reused on consecutive calls   to
-    \FUNC{shmem\_barrier} if the same active \ac{PE} set is used.
+    \FUNC{shmem\_barrier} if the same active set is used.
 }
 
 \apireturnvalues{
     None.
 }
 
 \apinotes{
-    If the \VAR{pSync} array is initialized at run time, be sure to use some type of
-    synchronization, for example, a call to \FUNC{shmem\_barrier\_all}, before
-    calling \FUNC{shmem\_barrier} for the first time.
+    If the \VAR{pSync} array is initialized at run time, another method of
+    synchronization (e.g., \FUNC{shmem\_barrier\_all}) must be used before
+    the initial use of that \VAR{pSync} array by \FUNC{shmem\_barrier}.
```

Comment:
How about this ?

If the \VAR{pSync} array is initialized at the run time, all \acp{PE} must be 
synchronized before the first call to \FUNC{shmem\_barrier} to ensure the completion 
of the initialization. For synchronizing the \acp{PE}, an alternate method 
(e.g., \FUNC{shmem\_barrier\_all}) must be used.

If the committee thinks that the below phrase is a problem, we can remove it.
> to ensure the completion of the initialization

Reply:
@nspark Can you express your opinion on this ? Otherwise, I'm ready to merge this.

Reply:
It's an improvement. How about:
> If the `pSync` array is initialized at the run time, all PEs must be synchronized before the first call to `shmem_barrier` (e.g., by `shmem_barrier_all`) to ensure the array has been initialized by all PEs before it is used.

Reply:
:+1